src/ripple/rpc/handlers/BookOffers.cpp

```diff
@@ -52,22 +52,21 @@ Json::Value doBookOffers (RPC::Context& context)
     if (!context.params.isMember (jss::taker_gets))
         return RPC::missing_field_error (jss::taker_gets);
 
-    if (!context.params[jss::taker_pays].isObject ())
+    Json::Value const& taker_pays = context.params[jss::taker_pays];
+    Json::Value const& taker_gets = context.params[jss::taker_gets];
+
+    if (!taker_pays.isObjectOrNull ())
         return RPC::object_field_error (jss::taker_pays);
 
-    if (!context.params[jss::taker_gets].isObject ())
+    if (!taker_gets.isObjectOrNull ())
```

Comment:
I wasn't sufficiently motivated to change the error message here in case someone was depending on it.  But if anyone feels strongly that the error message should be improved for null values, I'm happy to do that.